packages/vite/src/node/plugins/worker.ts

```diff
@@ -185,6 +185,20 @@ export async function workerFileToUrl(
   return encodeWorkerAssetFileName(fileName, workerMap)
 }
 
+export function webWorkerPostPlugin(): Plugin {
+  return {
+    name: 'vite:worker-post',
+    resolveImportMeta(property, { chunkId, format }) {
+      // document is undefined in the worker, so we need to avoid it in iife
+      if (property === 'url' && format === 'iife') {
+        return `new URL('${chunkId}', self.location.href).href`
```

Comment:
```suggestion
        return 'self.location.href'
```
I just noticed that we should use `self.location.href` here. `location.href` returns the location of the worker and not the document.
https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/location

`new URL('${chunkId}', self.location.href).href` is be like `http://localhost:4173/assets/assets/worker.js` which is not correct, because `chunkId` is like `/assets/worker.js` and `self.location.href` is like `http://localhost:4173/assets/worker.js`.